### PR TITLE
kubeadm operator - inital-commit

### PR DESCRIPTION
**Pull request by fabriziopandini**
This PR creates the kubeadm operator folder, adds a kubebuilder project skeleton and a set of hacks to be user for creating a kubeadm-operator-verify prepull job

/assign @neolit123
/cc @ereslibre  
/cc @yagonobre

### CODE REVIEW

File: operator/hack/boilerplate.go.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/*
+
```
Comment (ereslibre) [inline operator/hack/boilerplate.go.txt]: Add regular Kubernetes authors and date

File: operator/hack/fetch_bins.sh [hunk 2]
```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bash
+# Copyright 2018 The Kubernetes Authors.
```
Comment (ereslibre) [inline operator/hack/fetch_bins.sh]:
```suggestion
# Copyright 2019 The Kubernetes Authors.
```

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,3 @@
+module k8s.io/kubeadm
+
+go 1.12
```
Comment (ereslibre) [inline go.mod]: Do we want this toplevel `go.mod`?

Comment (neolit123) [reply go.mod]: we should't need it, as long as the sub-folders have a go.mod.

File: operator/hack/fetch_bins.sh (hunk 2 above)
Comment (fabriziopandini) [reply operator/hack/fetch_bins.sh]: done

File: operator/hack/boilerplate.go.txt (hunk 1 above)
Comment (fabriziopandini) [reply operator/hack/boilerplate.go.txt]: this is the output of kubebuilder init, and it is used for autogenerated files only.

I guess that keeping out of the way regular Kubernetes authors and date simplifies management, but eventually we can fix it in follow up PRs

File: go.mod (hunk 3 above)
Comment (fabriziopandini) [reply go.mod]: We don't care about this file, but it is generated by go every time you do an operation involving go modules in kinder or in the operator folder

File: operator/hack/boilerplate.go.txt (hunk 1 above)
Comment (ereslibre) [reply operator/hack/boilerplate.go.txt]: Yup, I know what it is used for, thought having it with correct contents would be nice nevertheless, but nothing blocking.

**Comment by ereslibre** [comment]
Thanks @fabriziopandini!

/lgtm

File: go.mod (hunk 3 above)
Comment (neolit123) [reply go.mod]: > We don't care about this file, but it is generated by go every time you do an operation involving go modules in kinder or in the operator folder

that's not entirely true.
if you run `cd kinder && ./hack/verify-deps.sh` the file will not be generated.

`/` itself is not a module and we should avoid treating it as such.

Comment (neolit123) [reply go.mod]: what operations are causing the file generation?

Comment (fabriziopandini) [reply go.mod]: I find it added every time a new dependency is added

**Comment by neolit123** [comment]
looks like we merged the test-infra PR without enabling docker-in-docker.
missing parts are:
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-discovery.yaml#L8

and

https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-discovery.yaml#L33-L34

**Pull request merged**